sphinx/application.py

```diff
@@ -310,23 +311,34 @@ def _init_env(self, freshenv):
                     logger.info(_('failed: %s'), err)
                 self._init_env(freshenv=True)
 
-    def _init_builder(self, buildername):
-        # type: (unicode) -> None
+    def create_builder(self, buildername):
+        # type: (unicode) -> Builder
         if buildername is None:
             logger.info(_('No builder selected, using default: html'))
             buildername = 'html'
         if buildername not in self.builderclasses:
             raise SphinxError(_('Builder name %s not registered') % buildername)
 
         builderclass = self.builderclasses[buildername]
-        self.builder = builderclass(self)
+        return builderclass(self)
+
+    def _init_builder(self):
+        # type: () -> None
+        self.builder.set_environment(self.env)
+        self.builder.init()
```

Comment:
-0 for try/except. This interface changing is not a important I think.
However, it would be nice to have a comment that maintaining init arguments for compatibility.